packages/tre/src/plugins/core/errors/index.ts

```diff
@@ -196,36 +196,73 @@ function getSourceMappedStack(workerSrcOpts: SourceOptions[], error: Error) {
 // matching the schema below with the `MF-Experimental-Error-Stack` header set
 // to a truthy value, in order to display the pretty-error page.
 export const HEADER_ERROR_STACK = "MF-Experimental-Error-Stack";
-const ErrorSchema = z.object({
-  message: z.ostring(),
-  name: z.ostring(),
-  stack: z.ostring(),
-});
+
+export interface JsonError {
+  message?: string;
+  name?: string;
+  stack?: string;
+  cause?: JsonError;
+}
+export const JsonErrorSchema: z.ZodType<JsonError> = z.lazy(() =>
+  z.object({
+    message: z.ostring(),
+    name: z.ostring(),
+    stack: z.ostring(),
+    cause: JsonErrorSchema.optional(),
+  })
+);
+
+export function reviveError(
+  workerSrcOpts: SourceOptions[],
+  jsonError: JsonError
+): Error {
+  let cause: Error | undefined;
+  if (jsonError.cause !== undefined) {
+    cause = reviveError(workerSrcOpts, jsonError.cause);
+  }
+
+  // If this is one of hte built-in error types, construct an instance of that
+  let ctor = Error;
+  if (jsonError.name !== undefined && jsonError.name in globalThis) {
+    const maybeCtor = (globalThis as Record<string, unknown>)[jsonError.name];
```

Comment:
This seems like a bit of a security issue (although it's all running locally with presumably trusted code, so...). What happens if someone sets the `jsonError.name` to `eval`, and `jsonMessage` to some executable code? I think `maybeCtor` should probably be allow-listed to just `Error` and `TypeError`.